test/validation/api/pktio/lso.c

```diff
@@ -844,12 +856,27 @@ static int is_ipv4_test_pkt(odp_packet_t pkt)
 	return 1;
 }
 
-static void check_ipv4_hdr(odp_packet_t pkt, uint16_t seg_num, uint32_t hdr_len)
+static void update_ipv4_hdr(uint8_t *hdr, uint32_t hdr_len, uint32_t orig_pkt_len,
+			    odp_packet_t pkt, uint16_t seg_num, uint16_t seg_offset)
 {
 	(void)seg_num;
-	(void)hdr_len;
+	uint32_t l3_offset = ODPH_ETHHDR_LEN;
+	uint16_t frag_offset;
+	odph_ipv4hdr_t ip;
 
 	CU_ASSERT(odp_packet_has_error(pkt) == 0);
+
+	memcpy(&ip, &hdr[l3_offset], sizeof(ip));
+
+	frag_offset = odp_be_to_cpu_16(ip.frag_offset);
+	frag_offset += seg_offset / 8;
+	if (odp_packet_len(pkt) + seg_offset < orig_pkt_len)
+		frag_offset |= LSO_TEST_IPV4_FLAG_MF;
+	ip.tot_len = odp_cpu_to_be_16(odp_packet_len(pkt) - l3_offset);
+	ip.frag_offset = odp_cpu_to_be_16(frag_offset);
+	ip.chksum = 0;
+	ip.chksum = ~odp_chksum_ones_comp16(&ip, hdr_len - l3_offset);
+	memcpy(hdr + l3_offset, &ip, sizeof(ip));
```

Comment:
Nitpick: in the previous memcpy it was &hdr[l3_offset]. Pick one style.

Reply:
v2